src/views/clusteroverview/OverviewTab/metric-charts-card/utils/hooks/utils.ts

```diff
@@ -0,0 +1,34 @@
+import { PrometheusValue } from '@openshift-console/dynamic-plugin-sdk';
+
+import { humanizeBinaryBytes } from '../../../../../../utils/utils/humanize';
+import { METRICS } from '../constants';
+
+export const getPointDay = (point: { x: Date; y: number }) => point?.x?.getDay();
+
+export const getLargestValue = (data: PrometheusValue[]) => {
+  return data?.reduce((acc, point) => {
+    const currValue = Number(point?.[1]);
+    acc = currValue > acc ? currValue : acc;
+    return acc;
+  }, -1);
+};
+
+export const findUnit = (metric: string, largestValue: number) => {
+  switch (metric) {
+    case METRICS.MEMORY:
+    case METRICS.STORAGE:
+      return humanizeBinaryBytes(largestValue)?.unit;
+    case METRICS.VCPU_USAGE:
+    case METRICS.VM:
+    default:
+      return null;
+  }
+};
```

Comment:
why do we need a switch here? are there plans to other types?

Reply:
I don't know why I got so switch happy in this PR. I rarely use them, but managed to use several here. Fixed! =)